src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs

```diff
@@ -560,7 +573,7 @@ private AzureDiskEncryptionStatusContext getStatusSinglePass(VirtualMachine vm)
             AzureDiskEncryptionStatusContext status = this.GetStatusFromInstanceView(vm);
 
             // Get Data Disk status from extension for Native Disk VMs
-            if (isNativeDiskVM(vm))
+            if ( status.DataVolumesEncrypted != EncryptionStatus.NoDiskFound && isNativeDiskVM(vm))
```

Comment:
GetStatusFromInstanceView can return null, so this should likely be ```status?.DataVolumesEncrypted```

Reply:
GetStatusFromInstanceView never returns null.